### PR TITLE
Downstream Single-Cell RNA-seq analysis with RaceID, StemID and FateID

### PR DESCRIPTION
**Pull request by mtekman**

### CODE REVIEW

**Comment by mtekman** [comment]
do not review, this needs more content

**Comment by mtekman** [comment]
@bgruening @bebatut I have just finished the first draft. Can one of you please review this on Monday?

**Comment by bgruening** [comment]
oki doki!

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+
+> ### {% icon question %} Question
+>
+> Are these genes expressed where we expect?
+>
+> > ### {% icon solution %} Solution
+> >
+> > They appear to overlap `c6` which is in close proximity to `c1`. There are two reasons why this might be the case:
+> > 1. `c1` is a small cluster and noisy cluster surrounded by more stably defined neighbours.
+> > 2. The three genes are more differentially expressed in `c1` than in `c6` if we look at them as independent entities, but their combined effect tells us that together they describe `c6` better.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: I do not think it's clear where you see c1 and c6 here ... the user needs to relate this to the images above ... maybe mention this in the image description and maybe also mention the scratchbook feature of galaxy to put both images next to each other.

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+>        - *"Use Defaults?"*: `Yes`
+>
+{: .hands_on}
+
+![Lineage Computation Plots]({{site.baseurl}}{% link topics/transcriptomics/images/raceid_stemid_lineage.png %} "Lineage Tree and Branches of significance.")
+
+The first (top-left) plot shows a minimum spanning tree that summarises the most likely connections between clusters. The second plot (top-right) elucidates this lineage hierarchy by projecting the cells along the links given by StemID, ordering the cells along each link in such a way to suggest a time series, or *pseudotime* analysis of each link. This is important, because by ordering the cells by lineage pseudotime we can trace the up or down regulation of a gene as discrete time points. The interval between each timepoint is of course hard to accurately determine, but the order of events is still important.
+
+The third (bottom-left) lends a degree of significance between clusters, where the colour indicates the level of significance between clusters, and the width indicates the link score computed by **StemID**. The thick red link between clusters 2-4, 3-5, and 3-1-6 have therefore a high level of significance attached to each of these.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: lends?

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,690 @@
...
+> ### {% icon question %} Questions
+>
+> Based on the above plots, which cluster is likely to be the main progenitor of the others?
+> 
+> > ### {% icon solution %} Solution
+> >
+> > On first glance, it appears that `c2` would be the progenitor, giving rise to; mature type `c11`, `c4`, and `c9`; `c3` which is the progenitor to `c5` and `c1`; `c8`, which gives rise to `c12` that in turn generates `c10`.
+> > `{2, {8 {12 {10}}, 11, 4, 9, 3 {5, 1 { 6 } } } }` (in JSON format)
+> > 
+> > This is actually not the case, and we will see why with the following images.
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: while I like the Q, this sentence "This is actually not the case, and we will see why with the following images." which serves as motivation to the next images is hidden and do not motivate people that have not read the Q

So I would take this sentence out of the box.

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,690 @@
...
+> >
+> {: .solution}
+>
+{: .question}
+
+
+![Link scores]({{site.baseurl}}{% link topics/transcriptomics/images/raceid_stemid_linkscores.png %} "Link scores between branches.")
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: make also clear where these images come from, which tool ...

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,690 @@
...
+    tutorials:
+##        - scrna-introduction
+        - scrna-plates-batches-barcodes
+        - scrna-umis
+        - scrna_preprocessing
+
+time_estimation: 2H
+key_points:
+  - Clustering single-cell RNA-seq data is often noisy
+  - Understanding the heirarchal relationships between clusters provides an understanding of cell development
```
Comment (bgruening) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: heirarchal  -> hierarchal

StemID can be used for ...
FateID can be used for ....?

**Comment by mtekman** [comment]
@bgruening thank you so much for the review! I have made changes relating to all your suggestions

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,712 @@
...
+Here we will explore one branching point of interest; `c3` giving rise to `c1` and `c5`. Will we be able to find an up or down regulation of genes between these two branches?
+
+
+> ### {% icon hands_on %} Hands-on: Comparing trajectory paths 3 to 1, and 3 to 5
+>
+> 1. **Lineage Branch Analysis using StemID** {% icon tool %} with the following parameters:
+>    - {% icon param-file %} *"Input RDS"*: `outrdat` (output of **Lineage computation using StemID** {% icon tool %})
+>    - In *"StemID Branch Link Examine"*:
+>        - *"Perform StemID?"*: `Yes`
+>            - *"Cluster Number"*: `1`
```
Comment (bebatut) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: Is it cluster 1 here?

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,712 @@
...
+>            - *"Trajectory Path i, j, k"*: `1,3,5`
+>            - *"Use Defaults?"*: `Yes`
+>    - In *"FateID Branch Link Examine"*:
+>        - *"Perform FateID?"*: `No`
+>
+{: .hands_on}
+
+![Heatmap Cells]({{site.baseurl}}{% link topics/transcriptomics/images/raceid_stemid_cells.png %} "StemID Heatmap of Cluster 1 cell projections.")
+
+The vertical names along the heatmap are the cells from `c1` being projected on to the two intercluster links (3 to 1, 3 to 5) with scores for each cluster being shown for that cell.
```
Comment (bebatut) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: I do not understand what is represented in the heatmap and how it is related to the trajectory Lineage Analysis

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,712 @@
...
+
+> ### {% icon hands_on %} Hands-on
+>
+> 1. **Lineage Branch Analysis using StemID** {% icon tool %} with the following parameters:
+>    - {% icon param-file %} *"Input RDS"*: `outrdat` (output of **Lineage computation using StemID** {% icon tool %})
+>    - In *"StemID Branch Link Examine"*:
+>        - *"Perform StemID?"*: `No`
+>    - In *"FateID Branch Link Examine"*:
+>        - *"Perform FateID?"*: `Yes`
+>            - *"Cells from Clusters"*: `1,3,5`
```
Comment (bebatut) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: It may be useful to explain why this order is important

**Comment by bgruening** [comment]
@mtekman I don't understand the error. Can you try to merge master into this branch?

**Comment by mtekman** [comment]
@bgruening any idea why I can't seem to link `link topics/transcriptomics/tutorials/scrna_preprocessing/tutorial.md` or `link topics/transcriptomics/tutorials/scrna-umis/tutorial.md` ?

File: topics/transcriptomics/tutorials/scrna-raceid/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,919 @@
...
+
+<!-- TODO: Subset analysis to validate prior cell labelling -->
+<!-- TODO: Replace sort/unique with datamash -->
+
+# Introduction
+{:.no_toc}
+
+The data provided here as part of this tutorial analyses single-cell RNA-seq data from a study published by [Grün et.al](https://doi.org/10.1016/j.stem.2016.05.010) in 2016. The data was used to cluster cells from *Lgr5*-positive intestinal stem cells of C57BL6/J mice, with the aim of discovering distinct cell sub-populations and deriving a lineage tree between them to find out how these sub-populations relate (or are derived from) one another.
+
+The input data consists of a single count matrix consisting of ~21,000 genes (rows) and ~400 cells (columns), generated via [scRNA pre-processing methods](({{site.baseurl}}{% link topics/transcriptomics/tutorials/scrna_preprocessing/tutorial.md %})) using the [CelSeq2 protocol](({{site.baseurl}}{% link topics/transcriptomics/tutorials/scrna-umis/tutorial.md %})).
```
Comment (bebatut) [inline topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]:
```suggestion
The input data consists of a single count matrix consisting of ~21,000 genes (rows) and ~400 cells (columns), generated via [scRNA pre-processing methods]({{site.baseurl}}{% link topics/transcriptomics/tutorials/scrna_preprocessing/tutorial.md %}) using the [CelSeq2 protocol]({{site.baseurl}}{% link topics/transcriptomics/tutorials/scrna-umis/tutorial.md %}).
```

It should solve the link issue

Comment (mtekman) [reply topics/transcriptomics/tutorials/scrna-raceid/tutorial.md]: thank you, could not spot that!

**Comment by bgruening** [comment]
Danke @mtekman and @bebatut, this is a great piece of work!

**Comment by bgruening** [comment]
@mtekman can you add this to singlecell.usegalaxy.eu?

**Pull request merged**